src/error.rs

```diff
@@ -1,41 +1,255 @@
+    ///
+    /// [`FromRequestError::Custom`]: enum.FromRequestError.html#variant.Custom
+    pub fn hyper_error(h: hyper::Error) -> Self
+    where
+        hyper::Error: Into<E>,
+    {
+        FromRequestError::Custom(h.into())
+    }
+
+    /// If this error is a [`FromRequestError::BuildIn`] return a reference to it.
+    ///
+    /// [`FromRequestError::BuildIn`]: struct.FromRequestError.html#variant.BuildIn
+    pub fn as_build_in(&self) -> Option<&BuildInError> {
+        use self::FromRequestError::*;
+
+        match self {
+            BuildIn(err) => Some(err),
+            _ => None,
+        }
+    }
+
+    /// Return the contained [`BuildInError`] if there is one.
+    ///
+    /// [`BuildInError`]: struct.BuildInError.html
+    pub fn into_build_in(self) -> Result<BuildInError, Self> {
+        use self::FromRequestError::*;
+
+        match self {
+            BuildIn(err) => Ok(err),
+            other => Err(other),
+        }
+    }
+
+    /// If this error is a [`FromRequestError::Custom`] return a reference to it.
+    ///
+    /// [`FromRequestError::Custom`]: struct.FromRequestError.html#variant.Custom
+    pub fn as_custom(&self) -> Option<&E> {
+        use self::FromRequestError::*;
+
+        match self {
+            Custom(err) => Some(err),
+            _ => None,
+        }
+    }
+
+    /// Return the contained custom error if there is one.
+    pub fn into_custom(self) -> Result<E, Self> {
+        use self::FromRequestError::*;
+
+        match self {
+            Custom(err) => Ok(err),
+            other => Err(other),
+        }
+    }
+
+    /// Turns this error into a generic boxed future compatible with the output
+    /// of `#[derive(FromRequest)]`.
+    ///
+    /// This is used by the code generated by `#[derive(FromRequest)]`.
+    #[doc(hidden)] // not part of public API
+    pub fn into_future<T: Send + 'static>(self) -> DefaultFuture<T, Self>
+    where
+        E: Send + 'static,
+    {
+        Box::new(Err(self).into_future())
+    }
+
+    /// Converts this error to a `FromRequestError<NewError>`.
+    ///
+    /// This is useful as transitive `From`/`Into` implementations tend to
+    /// run into conflicting implementations problems.
+    pub fn convert_custom_error<NewError>(self) -> FromRequestError<NewError>
+    where
+        E: Into<NewError>,
+    {
+        use self::FromRequestError::*;
+        match self {
+            Custom(err) => Custom(err.into()),
+            BuildIn(err) => BuildIn(err),
+        }
+    }
+}
+
+impl<E> StdError for FromRequestError<E>
+where
+    E: StdError + 'static,
+{
+    fn source(&self) -> Option<&(dyn StdError + 'static)> {
+        use self::FromRequestError::*;
+        match self {
+            Custom(err) => Some(err),
+            BuildIn(err) => Some(err),
+        }
+    }
+}
+
+impl<E> Display for FromRequestError<E>
+where
+    E: Display + 'static,
+{
+    fn fmt(&self, fter: &mut fmt::Formatter<'_>) -> fmt::Result {
```

Comment:
The rest of the codebase calls it `f`, so perhaps for consistency:

```suggestion
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
```